contracts/VeJoeStaking.sol

```diff
@@ -44,71 +37,43 @@ contract VeJoeStaking is Initializable, OwnableUpgradeable {
     /// @notice The upper limit of `maxCap`
     uint256 public upperLimitMaxCap;
 
-    /// @notice Rate of veJOE generated per sec per JOE staked, in parts per 1e18
-    uint256 public baseGenerationRate;
-
-    /// @notice Boosted rate of veJOE generated per sec per JOE staked, in parts per 1e18
-    uint256 public boostedGenerationRate;
+    /// @notice The accrued veJoe per share, scaled to `ACC_VEJOE_PER_SHARE_PRECISION`
+    uint256 public accVeJoePerShare;
 
-    /// @notice Precision of `baseGenerationRate` and `boostedGenerationRate`
-    uint256 public PRECISION;
+    /// @notice Precision of `accVeJoePerShare`
+    uint256 public ACC_VEJOE_PER_SHARE_PRECISION;
 
-    /// @notice Percentage of user's current staked JOE user has to deposit in order to start
-    /// receiving boosted benefits, in parts per 100.
-    /// @dev Specifically, user has to deposit at least `boostedThreshold/100 * userStakedJoe` JOE.
-    /// The only exception is the user will also receive boosted benefits if it's their first
-    /// time staking.
-    uint256 public boostedThreshold;
+    /// @notice The last time that the reward variables were updated
+    uint256 public lastRewardTimestamp;
 
-    /// @notice The length of time a user receives boosted benefits
-    uint256 public boostedDuration;
+    /// @notice Rate of veJOE generated per sec per JOE staked, scaled to `GENERATION_RATE_PRECISION`
+    uint256 public baseGenerationRate;
```

Comment:
If we're removing `boostedGenerationRate`, is there any point to having `baseGenerationRate`?

Reply:
Ah yes good point. In that case, let's rename to `veJoePerSec`